main.go

```diff
@@ -0,0 +1,227 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/icinga/check-netgear/internal/checks"
+	"github.com/icinga/check-netgear/netgear"
+	"os"
+	"slices"
+	"strconv"
+	"strings"
+
+	"github.com/NETWAYS/go-check"
+	"github.com/NETWAYS/go-check/result"
+)
+
+// So that flag supports slices
+type stringSliceFlag []string
+
+func (s *stringSliceFlag) String() string { return strings.Join(*s, ",") }
+func (s *stringSliceFlag) Set(v string) error {
+	*s = append(*s, v)
+	return nil
+}
+
+type intSliceFlag []int
+
+func (i *intSliceFlag) String() string {
+	parts := make([]string, 0, len(*i))
+	for _, v := range *i {
+		parts = append(parts, strconv.Itoa(v))
+	}
+	return strings.Join(parts, ",")
+}
+func (i *intSliceFlag) Set(v string) error {
+	n, err := strconv.Atoi(v)
+	if err != nil {
+		return err
+	}
+	*i = append(*i, n)
+	return nil
+}
+
+func main() {
+	hidecpu := flag.Bool("nocpu", false, "Hide the CPU info")
+	hidemem := flag.Bool("nomem", false, "Hide the RAM info")
+	hidetemp := flag.Bool("notemp", false, "Hide the Temperature info")
+	hidefans := flag.Bool("nofans", false, "Hide the Fans info")
+
+	var mode stringSliceFlag = []string{"basic"}
+	flag.Var(&mode, "mode", "Output modes to enable {basic|ports|poe|all} (repeatable)")
+
+	baseURL := flag.String("base-url", "http://192.168.0.239", "Base URL to use")
+
+	username := flag.String("username", "", "Username for authentication")
+	password := flag.String("password", "", "Password for authentication")
+
+	// Thresholds
+	cpuWarn := flag.Float64("cpu-warning", 50, "CPU usage warning threshold")
+	cpuCrit := flag.Float64("cpu-critical", 90, "CPU usage critical threshold")
+	memWarn := flag.Float64("mem-warning", 50, "RAM usage warning threshold")
+	memCrit := flag.Float64("mem-critical", 90, "RAM usage critical threshold")
+	fanWarn := flag.Float64("fan-warning", 3000, "Fan speed warning threshold")
+	tempWarn := flag.Float64("temp-warning", 50, "Temperature warning threshold")
+	tempCrit := flag.Float64("temp-critical", 70, "Temperature critical threshold")
+	statsWarn := flag.Float64("stats-warning", 5, "Port stats warning threshold")
+	statsCrit := flag.Float64("stats-critical", 20, "Port stats critical threshold")
+
+	var portsToCheck intSliceFlag = []int{1, 2, 3, 4, 5, 6, 7, 8}
+	flag.Var(&portsToCheck, "port", "Ports to check (repeatable)")
+
+	help := flag.Bool("help", false, "Show this help")
+	flag.BoolVar(help, "h", false, "Show this help (shorthand)")
+
+	flag.Parse()
+
+	if *help || *username == "" || *password == "" {
+		flag.Usage()
+		return
+	}
```

Comment:
I would split these up in two cases. If someone request the help, it's okay. However, if someone forgets to set credentials - or they somehow get lost along the way -, it is an error.

```suggestion
	if *help {
		flag.Usage()
		return
	}

	if *username == "" || *password == "" {
		fmt.Println("Both username and password are required")
		os.Exit(check.Unknown)
	}
```